xmake/modules/package/tools/cmake.lua

```diff
@@ -419,15 +419,24 @@ function _get_configs_for_android(package, configs, opt)
     local ndk = get_config("ndk")
     if ndk and os.isdir(ndk) then
         local ndk_sdkver = get_config("ndk_sdkver")
-        local ndk_cxxstl = get_config("ndk_cxxstl")
         table.insert(configs, "-DCMAKE_TOOLCHAIN_FILE=" .. path.join(ndk, "build/cmake/android.toolchain.cmake"))
+        table.insert(configs, "-DANDROID_USE_LEGACY_TOOLCHAIN_FILE=OFF")
         table.insert(configs, "-DANDROID_ABI=" .. package:arch())
         if ndk_sdkver then
             table.insert(configs, "-DANDROID_PLATFORM=android-" .. ndk_sdkver)
             table.insert(configs, "-DANDROID_NATIVE_API_LEVEL=" .. ndk_sdkver)
         end
-        if ndk_cxxstl then
-            table.insert(configs, "-DANDROID_STL=" .. ndk_cxxstl)
+        -- https://cmake.org/cmake/help/latest/variable/CMAKE_ANDROID_STL_TYPE.html
+        local supported_runtimes = {
+            "gnustl_static", "gnustl_shared",
+            "c++_static", "c++_shared",
+            "stlport_static", "stlport_shared"
+        }
+        for _, runtime in ipairs(supported_runtimes) do
+            if package:has_runtime(runtime) then
+                table.insert(configs, "-DCMAKE_ANDROID_STL_TYPE=" .. runtime)
+                break
+            end
         end
```

Comment:
```lua
local runtime = package:runtimes()
if runtime then
    table.insert(configs, "-DCMAKE_ANDROID_STL_TYPE=" .. runtime)
end
```

Reply:
what if multiple runtimes have been set?

Reply:
only one runtime. https://github.com/xmake-io/xmake/blob/d1ee61015c30cd08455b8d9bf85970b4ecaa079a/xmake/toolchains/ndk/xmake.lua#L36

```lua
set_runtimes("c++_static", "MT")
```

-> package:runtimes() only return `c++_static`

Reply:
I got confused by the plural form (runtimes instead of runtime).

Reply:
For some toolchains and platforms that support multiple runtimes, it may have multiple values, but for android/msvc, only one value is possible